### PR TITLE
Adding type_traits as the more common name

### PR DESCRIPTION
**Pull request by kunaltyagi**
> Can we change the name of `point_traits.h` to `type_traits.h`? Will this be a breaking change? if so, should we have a redirect to the new name?

At first sight yes, yes and yes.

_Originally posted by @SergioRAgostinho in https://github.com/PointCloudLibrary/pcl/pull/3651#issuecomment-592417114_

### CODE REVIEW

**Comment by kunaltyagi** [comment]
Yes. Deprecation between minor release doesn't sit right by me.

**Comment by aPonza** [comment]
Sorry to bother in a seemingly unrelated issue, but I think this header could be kept/subdivided. Let me explain. From https://github.com/PointCloudLibrary/pcl/pull/3701#issue-382337073 I have, among the other questions, this:

> - [ ] 16x "warning: Detected potential recursive class relation between class <x> and base class <y>" can be fixed with a `/** \cond NO_WARN_RECURSIVE */`-`/** \endcond */` pair for all four traits (as per [this SO answer](https://stackoverflow.com/a/33055550/9926122)) but it'd need a new header: just adding the pair leads to "warning: Conditional section with label 'NO_WARN_RECURSIVE' does not have a corresponding \endcond command within this file.". The solutions seems to be moving 
> https://github.com/PointCloudLibrary/pcl/blob/dac8961c36b7b575b6908adc7c6bf185ff631834/common/include/pcl/point_traits.h#L130-L187 and https://github.com/PointCloudLibrary/pcl/blob/dac8961c36b7b575b6908adc7c6bf185ff631834/common/include/pcl/register_point_struct.h#L316-L345 and https://github.com/PointCloudLibrary/pcl/blob/dac8961c36b7b575b6908adc7c6bf185ff631834/common/include/pcl/register_point_struct.h#L351-L356 into a new header with an appropriate name. Do I have the ok and which name would you like?

I'm thinking these bits I need from point_traits.h could be kept where they are and integrated with the register_point_struct.h bits. The rest of the point_traits.h code should then be moved to type_traits.h together with an `#include <pcl/point_traits.h>`.

Am I incorrectly understanding the code? Would you rather have a different header (point_struct_traits.h) or fix?

File: common/include/pcl/type_traits.h [hunk 1]
```diff
@@ -0,0 +1,359 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2010-2012, Willow Garage, Inc.
```
Comment (kunaltyagi) [inline common/include/pcl/type_traits.h]: Is a new copyright year needed?

Comment (SergioRAgostinho) [reply common/include/pcl/type_traits.h]: Yes.

**Comment by kunaltyagi** [comment]
I'll rebase it a bit first.

**Pull request merged**